mda_openbabel_converter/OpenBabelParser.py

```diff
@@ -4,26 +4,119 @@
+except ImportError:
+    print("Cannot find openbabel, install with 'pip install openbabel==2.4.0'")
 
 class OpenBabelParser():
     """
     Inherits from TopologyReaderBase and converts an OpenBabel OBMol to a 
     MDAnalysis Topology or adds it to a pre-existing Topology. This parser will 
     does not work in the reverse direction.
     """
+
+    @staticmethod
     def _format_hint(thing):
         """
         Base function to check if the parser can actually parse this “thing” 
-        (i.e., is it a valid OpenBabel OBMol with no missing information, that 
-        can be converted to a MDAnalysis Topology?)
+        (i.e., is it a valid OpenBabel OBMol that can be converted to a 
+        MDAnalysis Topology?)
         """
-        pass
+        try:
+            import openbabel as OB
+        except ImportError:
+            return False
+        else:
+            return isinstance(thing, OB.OBMol)
 
-    def parse(self, **kwargs):
+    def parse(self, filename: OBMol, **kwargs):
         """
         Accepts an OpenBabel OBMol and returns a MDAnalysis Topology. Will need
         to extract the number of atoms, number of residues, number of segments,
-        atom_residue index, residue_segment index and other attributes from the
-        OBMol to initialise a new Topology. 
+        atom_residue index, residue_segment index and all of the atom's 
+        relevant attributes from the OBMol to initialise a new Topology. 
         """
+        format = 'OPENBABEL'
+
+        self.atoms = []
+        self.n_atoms = 0
+        self.residues = []
+        self.n_residues = 0
+        self.segments = []
+        self.n_segments = 0
+
+        obmol = filename
+
+        # Atoms
+        names = []
+        chiralities = []
+        resnums = []
+        resnames = []
+        elements = []
+        masses = []
+        charges = []
+        aromatics = []
+        ids = []
+        atomtypes = []
+        segids = []
+        altlocs = []
+        chainids = []
+        icodes = []
+        occupancies = []
+        tempfactors = [] # B factor; not supported by OB
+
+        if obmol.GetFirstAtom().equals(None):
+            return Topology(n_atoms = 0,
+                            n_res = 0,
+                            n_seg = 0,
+                            attrs=None,
+                            atom_resindex=None,
+                            residue_segindex=None)
+
+        for atom in OB.OBMolAtomIter(obmol):
+            # need to add handling incase attributes are invalid or null in OBMol
+            # names.append(atom.GetType()) #char -> nothing for name in OBMol? Is name required to make MDA Atom?
+            atomtypes.append(atom.GetType()) #char
+            ids.append(atom.GetIdx()) #int
+            masses.append(atom.GetExactMass()) #double -> what about atom.GetAtomicMass()??; which is better?
+            if not atom.GetExactMass().equals(atom.GetAtomicMass()):
+                warnings.warn(
```

Comment:
Are they not? If only the mass is different should be ok? We need to write a test to check this.

Reply:
I'm a bit confused. I think we discussed that MDAnalysis doesn't support/show isotopes so it's worth giving a warning. And isotopes would have a mass difference? Otherwise, I can use "atom.HasIsotope" from OpenBabel instead, but I figured manually checking the masses may be safer? Would said test be to check if there are differences between the masses?

Reply:
MDAnalysis does not have a *specific* way to flag isotopes, but it does have an array of masses. Can we not just insert the exact mass into the array? This would be the closest translation IMO.

Reply:
By array of masses, do you mean that there is an array for each Atom? I have already saved the exact mass (line 88), which will later be assigned to the atom. Is that what you mean?

Reply:
Yep that is what I mean, that list gets converted to a `numpy.ndarray` later with one mass for each atom.

Reply:
I peeked at the docs and I agree `GetExactMass()` looks like the right one. I think it might be more clear to edit the `warnings.warn` comment in line 114 to say that MDAnalysis does not flag (rather than support) isotopes and to encourage the user to check the exact masses themselves. Thoughts?